Common/TableProducer/trackextension.cxx

```diff
@@ -32,21 +36,44 @@ using namespace o2::framework::expressions;
  */
 //****************************************************************************************
 struct TrackExtensionTask {
+  Configurable<int> cfgDcaMethod{"dcamethod", 1, "Method to estimate the track DCA: 0 = crude, 1 = minimum, 2 = rigorous. Default minimum"};
 
   Produces<aod::TracksExtended> extendedTrackQuantities;
 
   void process(aod::FullTracks const& tracks, aod::Collisions const&)
   {
+    o2::base::Propagator::MatCorrType matCorr = o2::base::Propagator::MatCorrType::USEMatCorrLUT;
+    if ((cfgDcaMethod == 1) or (cfgDcaMethod == 2)) {
+      o2::base::GeometryManager::loadGeometry();
+      o2::base::Propagator::initFieldFromGRP();
+      if (cfgDcaMethod == 2) {
+        auto matLUTFile = o2::base::NameConf::getMatLUTFileName();
+        if (o2::utils::Str::pathExists(matLUTFile)) {
+          auto* lut = o2::base::MatLayerCylSet::loadFromFile(matLUTFile);
+          o2::base::Propagator::Instance()->setMatLUT(lut);
+        }
+      }
+    }
     for (auto& track : tracks) {
 
       std::array<float, 2> dca{1e10f, 1e10f};
       if (track.has_collision()) {
         if ((track.trackType() == o2::aod::track::TrackTypeEnum::Track) ||
             (track.trackType() == o2::aod::track::TrackTypeEnum::Run2Track && track.itsChi2NCl() != 0.f && track.tpcChi2NCl() != 0.f && std::abs(track.x()) < 10.f)) {
-          float magField = 5.0; // in kG (FIXME: get this from CCDB)
           auto trackPar = getTrackPar(track);
           auto const& collision = track.collision();
-          trackPar.propagateParamToDCA({collision.posX(), collision.posY(), collision.posZ()}, magField, &dca);
+          if (cfgDcaMethod == 1) {
+            trackPar.propagateParamToDCA({collision.posX(), collision.posY(), collision.posZ()}, o2::base::Propagator::Instance()->getNominalBz(), &dca);
+          } else if (cfgDcaMethod == 2) {
+            o2::dataformats::DCA dcaInfo;
+            if (o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackPar, 2., matCorr, &dca)) {
+              dca[0] = dcaInfo.getY();
+              dca[1] = dcaInfo.getZ();
+            }
```

Comment:
From Ruben

you are passing to `propagateToDCABxByBz` the `dca` then overwrite it with `dcaInfo` which was never filled.